### PR TITLE
Better fits

### PR DESCRIPTION
**Pull request by gribeill**
New API for fits in the analysis code:

 - New `AuspexFit` class that wraps `scipy.optimize.curve_fit` and gives easier access to model function, fit results and errors, and goodness-of-fit statistics. 
 - Re-write qubit fits using new style and update `pulse_calibrations`.
 - Add Lorentzian fit method.
 - Add tests for commonly used fits
 - Merge in @BillyKalfus's resonator circle fitting code.
 - Split `auspex.analysis.fits.py` into multiple files for readability.

### CODE REVIEW

File: src/auspex/analysis/qubit_fits.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    """
+    xlabel = "Delay"
+    ylabel = r"<$\sigma_z$>"
+    title = r"$T_1$ Fit"
+
+    @staticmethod
+    def _model(x, *p):
+        return p[0]*np.exp(-x/p[1]) + p[2]
+
+    def _initial_guess(self):
```
Comment (caryan) [inline src/auspex/analysis/qubit_fits.py]: Random aside : the method [here](https://www.scribd.com/doc/14674814/Regressions-et-equations-integrales) (pg. 17) is super slick for getting the exponential decay guess. As a bonus @gribeill  should be able to read it :grinning:.

File: src/auspex/analysis/helpers.py [hunk 2]
```diff
@@ -17,3 +68,89 @@ def normalize_data(data, zero_id = 0, one_id = 1):
...
+    Parameters
+    ----------
+    data : unscaled data with cal points
+
+    Returns
+    -------
+    data : scaled data array
+    """
+    # assume with have 2 cal repeats
+    # TO-DO: make this general!!
```
Comment (matthewware) [inline src/auspex/analysis/helpers.py]: As a total aside, we should actually attempt this for multi-qubit data.

Comment (dieris) [reply src/auspex/analysis/helpers.py]: not perfect, but I've been using this for any number of qubits: https://github.com/BBN-Q/Qlab.jl/commit/3732123def8903696796d466ada9dad765f55bb4

**Comment by matthewware** [comment]
Looks good to me.  I haven't poked at all the code myself though.  If I can get to them before the merge I might have a few things to add.  If you're ready to merge, punch it.

File: src/auspex/analysis/qubit_fits.py (hunk 1 above)
Comment (gribeill) [reply src/auspex/analysis/qubit_fits.py]: Oh that looks like a very interesting reference... merci @caryan !

**Comment by matthewware** [comment]
The 4.5 Miniconda build stalled for some reason.  Feel free to drop that commit if you're in a hurry.  Not sure why only that test would fail though...

**Comment by matthewware** [comment]
🎊 🎊

**Pull request merged**